src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala

```diff
@@ -198,7 +198,7 @@ class SyncController(
       else
         (SyncState.empty, fastSyncStateStorage.purge())
 
-    private var mptNodesQueue: Set[HashType] = initialSyncState.mptNodesQueue
+    private var mptNodesQueue: Seq[HashType] = initialSyncState.mptNodesQueue
     private var nonMptNodesQueue: Set[HashType] = initialSyncState.nonMptNodesQueue
     private var blockBodiesQueue: Set[ByteString] = initialSyncState.blockBodiesQueue
     private var receiptsQueue: Set[ByteString] = initialSyncState.receiptsQueue
```

Comment:
Well, this code bit is fully isolated, there is no interface, there are no callers. So we can explicitly choose and enforce the implementation that has the most fitting performance characteristics. 

And we shouldn't rely on the fact that a default `Seq` is a `List` because it may change in the future Scala releases. It used to be a `Vector`.

Now, it does make some theoretical sense to use a supertype in a parameter position like in line 390. Probably less so in practice though, personally I would also change it to `List` (if we wanted to be really permissive we should probably use something like `Traversable`).